drivers/display/display_led_strip_matrix.c

```diff
@@ -0,0 +1,255 @@
+				   const uint16_t y, const struct display_buffer_descriptor *desc)
+{
+	__ASSERT(desc->width <= desc->pitch, "Pitch is smaller then width");
+	__ASSERT(desc->pitch <= config->width, "Pitch in descriptor is larger than screen size");
+	__ASSERT(desc->height <= config->height, "Height in descriptor is larger than screen size");
+	__ASSERT(x + desc->pitch <= config->width,
+		 "Writing outside screen boundaries in horizontal direction");
+	__ASSERT(y + desc->height <= config->height,
+		 "Writing outside screen boundaries in vertical direction");
+
+	if (desc->width > desc->pitch || x + desc->pitch > config->width ||
+	    y + desc->height > config->height) {
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int led_strip_matrix_write(const struct device *dev, const uint16_t x, const uint16_t y,
+				  const struct display_buffer_descriptor *desc, const void *buf)
+{
+	const struct led_strip_matrix_config *config = dev->config;
+	const uint32_t *pixel_buf_ptr = buf;
+	int rc;
+
+	rc = check_descriptor(config, x, y, desc);
+	if (rc) {
+		LOG_ERR("Invalid descriptor: %d", rc);
+		return rc;
+	}
+
+	for (size_t ypos = y; ypos < (y + desc->height); ypos++) {
+		for (size_t xpos = x; xpos < (x + desc->width); xpos++) {
+			uint32_t color = *pixel_buf_ptr++;
+			struct led_rgb *pix = pixel_address(config, xpos, ypos);
+
+			pix->r = (color >> 16) & 0xFF;
+			pix->g = (color >> 8) & 0xFF;
+			pix->b = (color) & 0xFF;
+		}
+		pixel_buf_ptr += desc->pitch - desc->width;
+	}
+
+	for (size_t i = 0; i < config->num_of_strips; i++) {
+		rc = led_strip_update_rgb(config->strips[i].dev, config->strips[i].pixels,
+					  config->width * config->height);
+		if (rc) {
+			LOG_ERR("couldn't update strip: %d", rc);
+		}
+	}
+
+	return rc;
+}
+
+static int led_strip_matrix_read(const struct device *dev, const uint16_t x, const uint16_t y,
+				 const struct display_buffer_descriptor *desc, void *buf)
+{
+	const struct led_strip_matrix_config *config = dev->config;
+	uint32_t *pixel_buf_ptr = buf;
+	int rc;
+
+	rc = check_descriptor(config, x, y, desc);
+	if (rc) {
+		LOG_ERR("Invalid descriptor: %d", rc);
+		return rc;
+	}
+
+	for (size_t ypos = y; ypos < (y + desc->height); ypos++) {
+		for (size_t xpos = x; xpos < (x + desc->width); xpos++) {
+			struct led_rgb *pix = pixel_address(config, xpos, ypos);
+
+			*pixel_buf_ptr = 0xFF000000 | pix->r << 16 | pix->g << 8 | pix->b;
+			pixel_buf_ptr++;
+		}
+		pixel_buf_ptr += desc->pitch - desc->width;
+	}
+
+	return 0;
+}
+
+static void led_strip_matrix_get_capabilities(const struct device *dev,
+					      struct display_capabilities *caps)
+{
+	const struct led_strip_matrix_config *config = dev->config;
+
+	memset(caps, 0, sizeof(struct display_capabilities));
+	caps->x_resolution = config->width;
+	caps->y_resolution = config->height;
+	caps->supported_pixel_formats = PIXEL_FORMAT_ARGB_8888;
+	caps->current_pixel_format = PIXEL_FORMAT_ARGB_8888;
```

Comment:
btw was there a reason for not using `PIXEL_FORMAT_RGB_888`? It doesn't look like alpha channel can be used in any meaningful way so having 24-bit per colour instead of 32 could save quite a bit of memory.

Reply:
I changed to be able to select pixel format on devicetree.